tests/components/hassio/test_update.py

```diff
@@ -746,6 +860,43 @@ async def test_update_os_with_error(
         )
 
 
+async def test_update_os_with_backup_and_error(
+    hass: HomeAssistant,
+    supervisor_client: AsyncMock,
+) -> None:
+    """Test updating OS update entity with error."""
+    config_entry = MockConfigEntry(domain=DOMAIN, data={}, unique_id=DOMAIN)
+    config_entry.add_to_hass(hass)
+
+    with patch.dict(os.environ, MOCK_ENVIRON):
+        result = await async_setup_component(
+            hass,
+            "hassio",
+            {"http": {"server_port": 9999, "server_host": "127.0.0.1"}, "hassio": {}},
+        )
+        assert result
+    await setup_backup_integration(hass)
+
+    supervisor_client.os.update.return_value = None
+    supervisor_client.mounts.info.return_value.default_backup_mount = None
+    with (
+        patch(
+            "homeassistant.components.backup.manager.BackupManager.async_create_backup",
+            side_effect=BackupManagerError,
+        ),
+        pytest.raises(HomeAssistantError, match=r"^Error creating backup:"),
+    ):
+        assert not await hass.services.async_call(
```

Comment:
```suggestion
        await hass.services.async_call(
```

Reply:
I think the intention here is to assert that the service call fails, but that's not correct because a service call without a response just returns `None`?
If so, this is wrong also in the existing tests.

Reply:
Yeah, the return value of this call doesn't mean that the service call failed.